controller/eks-cluster-config-handler.go

```diff
@@ -793,27 +792,69 @@ func BuildUpstreamClusterState(name string, clusterState *eks.DescribeClusterOut
 			continue
 		}
 		ngToAdd := eksv1.NodeGroup{
-			NodegroupName: ng.Nodegroup.NodegroupName,
-			DiskSize:      ng.Nodegroup.DiskSize,
-			Labels:        ng.Nodegroup.Labels,
-			DesiredSize:   ng.Nodegroup.ScalingConfig.DesiredSize,
-			MaxSize:       ng.Nodegroup.ScalingConfig.MaxSize,
-			MinSize:       ng.Nodegroup.ScalingConfig.MinSize,
-			Subnets:       aws.StringValueSlice(ng.Nodegroup.Subnets),
-			Tags:          ng.Nodegroup.Tags,
-			Version:       ng.Nodegroup.Version,
-		}
-		if len(ng.Nodegroup.InstanceTypes) > 0 {
-			ngToAdd.InstanceType = ng.Nodegroup.InstanceTypes[0]
+			NodegroupName:        ng.Nodegroup.NodegroupName,
+			DiskSize:             ng.Nodegroup.DiskSize,
+			Labels:               ng.Nodegroup.Labels,
+			DesiredSize:          ng.Nodegroup.ScalingConfig.DesiredSize,
+			MaxSize:              ng.Nodegroup.ScalingConfig.MaxSize,
+			MinSize:              ng.Nodegroup.ScalingConfig.MinSize,
+			Subnets:              aws.StringValueSlice(ng.Nodegroup.Subnets),
+			Tags:                 ng.Nodegroup.Tags,
+			Version:              ng.Nodegroup.Version,
+			RequestSpotInstances: aws.Bool(aws.StringValue(ng.Nodegroup.CapacityType) == eks.CapacityTypesSpot),
+		}
+
+		if aws.BoolValue(ngToAdd.RequestSpotInstances) {
+			ngToAdd.SpotInstanceTypes = ng.Nodegroup.InstanceTypes
+		}
+
+		if ng.Nodegroup.LaunchTemplate != nil {
+			if managedTemplateID == aws.StringValue(ng.Nodegroup.LaunchTemplate.Id) {
+				// If this is a rancher-managed launch template, then we move the data from the launch template to the node group.
+				launchTemplateRequestOutput, err := ec2Service.DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
+					LaunchTemplateId: ng.Nodegroup.LaunchTemplate.Id,
+					Versions:         []*string{ng.Nodegroup.LaunchTemplate.Version},
+				})
+				if err != nil || len(launchTemplateRequestOutput.LaunchTemplateVersions) == 0 {
```

Comment:
Do we want to return or report the error in some way here? It's ignored currently, which may be the right idea, but wanted to make sure.

Reply:
Yes, this has changed in the next update to this PR.